kroxylicious/src/test/java/io/kroxylicious/proxy/internal/filter/ConstructorInjection.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.internal.filter;
+
+import io.kroxylicious.proxy.filter.Filter;
+import io.kroxylicious.proxy.filter.FilterFactory;
+import io.kroxylicious.proxy.filter.FilterFactoryContext;
+import io.kroxylicious.proxy.plugin.PluginConfigType;
+import io.kroxylicious.proxy.plugin.PluginConfigurationException;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@PluginConfigType(ConstructorInjectionConfig.class)
+public class ConstructorInjection implements FilterFactory<ConstructorInjectionConfig, String> {
```

Comment:
We've got some FilterFactory implementations called X, some XFactory, and some XFilterFactory.   I prefer the latter.

Reply:
In the discussion we've agreed that:

1.  In the configuration the `type` will refer to the factory class, not the filter class.
2. That the person doing the configuring really only cares about naming the _functionality_, so adopting the convention that every factory should be suffixed withe `FilterFactory` is just making their experience worse. 
3. Therefore I intend, once this PR is merged, to rename all the existing `XFilterFactory` classes to `X`. 
4. So I'm a bit loathe to change this in this PR, only to change it it in the next.

Reply:
I'm happy with that.